forge-game/src/main/java/forge/game/ability/effects/RingTemptsYouEffect.java

```diff
@@ -0,0 +1,145 @@
+package forge.game.ability.effects;
+
+import forge.GameCommand;
+import forge.ImageKeys;
+import forge.card.CardRarity;
+import forge.card.MagicColor;
+import forge.game.Game;
+import forge.game.ability.AbilityFactory;
+import forge.game.ability.AbilityKey;
+import forge.game.card.Card;
+import forge.game.card.CardCollection;
+import forge.game.player.Player;
+import forge.game.spellability.AbilitySub;
+import forge.game.spellability.SpellAbility;
+import forge.game.trigger.Trigger;
+import forge.game.trigger.TriggerHandler;
+import forge.game.trigger.TriggerType;
+import forge.game.zone.ZoneType;
+
+import java.util.Map;
+
+public class RingTemptsYouEffect extends EffectEffect {
+
+    @Override
+    protected String getStackDescription(SpellAbility sa) {
+        final StringBuilder sb = new StringBuilder();
+        sb.append("The Ring tempts " + sa.getActivatingPlayer() + ".");
+        return sb.toString();
+    }
+
+    @Override
+    public void resolve(SpellAbility sa) {
+        Player p = sa.getActivatingPlayer();
+        Game game = p.getGame();
+        Card card = sa.getHostCard();
+        Card theRing = null;
+        for (Card c : p.getCardsIn(ZoneType.Command)) {
+            if (c.getName().equals("The Ring")) {
+                theRing = c;
+                break;
+            }
+        }
+
+        int level = 0;
+        if (theRing == null) {
+            // Create the ring with first level
+            String image = ImageKeys.getTokenKey("the_ring");
+            theRing = createEffect(sa, p, "The Ring", image);
+            theRing.setSetCode(card.getSetCode());
+            theRing.setRarity(CardRarity.Common);
+            theRing.setColor(MagicColor.COLORLESS);
+
+            game.getTriggerHandler().suppressMode(TriggerType.ChangesZone);
+            theRing = p.getGame().getAction().moveTo(p.getZone(ZoneType.Command), theRing, sa);
+            game.getTriggerHandler().clearSuppression(TriggerType.ChangesZone);
+        } else {
+            level = Integer.parseInt(theRing.getSVar("RingLevel"));
+        }
+
+        level += 1;
+        theRing.setSVar("RingLevel", String.valueOf(Math.min(level, 4)));
+
+        switch(level) {
+            case 1:
+                String legendary = "Mode$ Continuous | EffectZone$ Command | Affected$ Card.IsRemembered+YouCtrl | AddType$ Legendary | AddDesignation$ Ring-bearer | Description$ Your Ring-bearer is legendary.";
```

Comment:
- We could keep Designations in Card to later refactor stuff like Monstrous or Renowned into it?
- it should still just be set directly after `chooseSingleEntity`
- these effects can just use `IsRingBearer` property
- dont need extra ringBearerCommands
  - it should remove designation via loseControlCommands
  - Ring effect might want to keep Remembered so it can find the old bearer to remove when choosing new one

Reply:
The Ring Effect should not keep the old bearer

we even could refactor to have the ring bearer be stored in the Player Object, then we wouldn't need to remember at all